icx-proxy/src/main.rs

```diff
@@ -459,8 +472,11 @@ async fn handle_request(
                 .build()
                 .expect("Could not create agent..."),
         );
-
-        forward_request(request, agent, dns_canister_config.as_ref(), logger.clone()).await
+        if fetch_root_key && agent.fetch_root_key().await.is_err() {
```

Comment:
fetch_root_key is meant to be called once, after constructing the agent, rather than many times preceding individual operations.